hs_app_netCDF/tests/views/test_create_netcdf_resource.py

```diff
@@ -0,0 +1,135 @@
+import os
+import tempfile
+import shutil
+
+from django.contrib.sessions.middleware import SessionMiddleware
+from django.test import TestCase, RequestFactory
+from django.contrib.auth.models import Group
+from django.contrib.messages.storage.fallback import FallbackStorage
+from django.core.urlresolvers import reverse
+
+from rest_framework import status
+
+from hs_core import hydroshare
+from hs_core.models import BaseResource, ResourceFile
+from hs_core.views import create_resource
+from hs_core.testing import MockIRODSTestCaseMixin
+
+
+class TestCreateResourceViewFunctions(MockIRODSTestCaseMixin, TestCase):
+    def setUp(self):
+        super(TestCreateResourceViewFunctions, self).setUp()
+        self.group, _ = Group.objects.get_or_create(name='Hydroshare Author')
+        self.username = 'john'
+        self.password = 'jhmypassword'
+        self.user = hydroshare.create_account(
+            'john@gmail.com',
+            username=self.username,
+            first_name='John',
+            last_name='Clarson',
+            superuser=False,
+            password=self.password,
+            groups=[]
+        )
+
+        self.factory = RequestFactory()
+
+        self.temp_dir = tempfile.mkdtemp()
+
+        self.netcdf_file_name = 'netcdf_valid.nc'
+        self.netcdf_file = 'hs_app_netCDF/tests/{}'.format(self.netcdf_file_name)
+        target_temp_netcdf_file = os.path.join(self.temp_dir, self.netcdf_file_name)
+        shutil.copy(self.netcdf_file, target_temp_netcdf_file)
+        self.netcdf_file_obj = open(target_temp_netcdf_file, 'r')
+
+        self.netcdf_bad_file_name = 'netcdf_invalid.nc'
+        self.netcdf_bad_file = 'hs_app_netCDF/tests/{}'.format(self.netcdf_bad_file_name)
+        target_temp_bad_netcdf_file = os.path.join(self.temp_dir, self.netcdf_bad_file_name)
+        shutil.copy(self.netcdf_bad_file, target_temp_bad_netcdf_file)
+        self.netcdf_bad_file_obj = open(target_temp_bad_netcdf_file, 'r')
+
+    def tearDown(self):
+        super(TestCreateResourceViewFunctions, self).tearDown()
+        if os.path.exists(self.temp_dir):
+            shutil.rmtree(self.temp_dir)
+
+    def test_create_resource(self):
+        # here we are testing the create_resource view function
+
+        # test with no file upload
+        post_data = {'resource-type': 'NetcdfResource',
+                     'title': 'Test NetCDF Resource Creation',
+                     'irods_federated': 'true'
+                     }
+        url = reverse('create_resource')
+        request = self.factory.post(url, data=post_data)
+        request.user = self.user
+        self._set_request_message_attributes(request)
+        self._add_session_to_request(request)
+
+        response = create_resource(request)
+        self.assertEqual(response.status_code, status.HTTP_302_FOUND)
+        res_id = response.url.split('/')[2]
+        self.assertEqual(BaseResource.objects.filter(short_id=res_id).exists(), True)
```

Comment:
assertTrue() is simpler and right to the point you are testing here. You should be able to put in more detailed message if it fails to provide more customized message when assert fails. But if you think assertEqual(x, True) works better for you, it is fine with me to keep it the way it is.